File: src/test/java/cube8540/oauth/authentication/credentials/oauth/client/application/DefaultOAuth2ClientDetailsServiceTest.java
package cube8540.oauth.authentication.credentials.oauth.client.application;

import cube8540.oauth.authentication.credentials.oauth.client.domain.OAuth2ClientRepository;
import cube8540.oauth.authentication.credentials.oauth.error.OAuth2ClientRegistrationException;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

import static cube8540.oauth.authentication.credentials.oauth.client.application.OAuth2ClientApplicationTestHelper.RAW_CLIENT_ID;
import static cube8540.oauth.authentication.credentials.oauth.client.application.OAuth2ClientApplicationTestHelper.makeEmptyClientRepository;
import static org.junit.jupiter.api.Assertions.assertThrows;

@DisplayName("OAuth2 클라이언트 디테일즈 서비스 테스트")
class DefaultOAuth2ClientDetailsServiceTest {

    @Test
    @DisplayName("저장소에 등롣 되지 않은 클라이언트 검색")
    void searchNotRegisteredClientInRepository() {
        OAuth2ClientRepository repository = makeEmptyClientRepository();
        DefaultOAuth2ClientDetailsService service = new DefaultOAuth2ClientDetailsService(repository);

        assertThrows(OAuth2ClientRegistrationException.class, () -> service.loadClientDetailsByClientId(RAW_CLIENT_ID));
    }

}

File: src/test/kotlin/cube8540/oauth/authentication/oauth/client/application/DefaultOAuth2ClientDetailsServiceTest.kt
package cube8540.oauth.authentication.oauth.client.application

import cube8540.oauth.authentication.oauth.client.domain.OAuth2ClientId
import cube8540.oauth.authentication.oauth.client.domain.OAuth2ClientRepository
import cube8540.oauth.authentication.oauth.error.OAuth2ClientRegistrationException
import io.mockk.every
import io.mockk.mockk
import org.assertj.core.api.Assertions.assertThat
import org.assertj.core.api.Assertions.catchThrowable
import org.junit.jupiter.api.Test
import java.util.*

class DefaultOAuth2ClientDetailsServiceTest {

    private val repository: OAuth2ClientRepository = mockk()

    private val service = DefaultOAuth2ClientDetailsService(repository)

    @Test
    fun `search not registered client in repository`() {
        every { repository.findByClientId(OAuth2ClientId("clientId")) } returns Optional.empty()

        val thrown = catchThrowable { service.loadClientDetailsByClientId("clientId") }
        assertThat(thrown).isInstanceOf(OAuth2ClientRegistrationException::class.java)
    }
}